### PR TITLE
Several terraform updates

### PR DESCRIPTION
**Pull request by aarontp**
Does a few things:
* Sets the project by default in all gcloud calls which weren't previously setting this
* Checks for gcloud/terraform binaries
* Uses the gcloud configured project if the cloudshell project environment variable is not set
* Uses native terraform configs for Datastore, removing the local-exec
* Does check for config file, and backs it up before writing new config.
* Sets infrastructure ID for GCS pubsub subscription so we can isolate installations
* Updates the Turbinia config template

Only downside of removing the local-exec for the Datastore is that it makes it harder to ignore when the indexes are already configured, so this will fail at start time if they are already configured.  Open to suggestions here.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -2,11 +2,43 @@
...
+if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+  DEVSHELL_PROJECT_ID=$(gcloud config get-value project)
+  ERRMSG="ERROR: Could not get configured project. Please either restart "
+  ERRMSG+="Google Cloudshell, or set configured project with "
+  ERRMSG+="'gcloud config set project PROJECT' when running outside of Cloudshell."
+  if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+  echo "Environment variable \$DEVSHELL_PROJECT_ID not set"
```
Comment (hacktobeer) [inline deploy.sh:31]: You check the env var in line 27 so it should always be set here.

File: deploy.sh [hunk 2]
```diff
@@ -60,24 +87,28 @@ elif [[ $( gcloud auth list --filter="status:ACTIVE" --format="value(account)" |
...
-  num_functions="$(gcloud functions list | grep task | wc -l)"
+  num_functions="$(gcloud functions --project $DEVSHELL_PROJECT_ID list | grep task | grep $TURBINIA_REGION | wc -l)"
   if [[ "${num_functions}" -eq "3" ]]; then
     echo "All Cloud Functions deployed"
     break
   fi
-  gcloud -q functions deploy gettasks --source modules/turbinia/data/ --runtime nodejs10 --trigger-http --memory 256MB --timeout 60s
-  gcloud -q functions deploy closetask --source modules/turbinia/data/ --runtime nodejs10 --trigger-http --memory 256MB --timeout 60s
-  gcloud -q functions deploy closetasks --source modules/turbinia/data/ --runtime nodejs10 --trigger-http --memory 256MB --timeout 60s
+  gcloud --project $DEVSHELL_PROJECT_ID -q functions deploy gettasks --region $TURBINIA_REGION --source modules/turbinia/data/ --runtime nodejs8 --trigger-http --memory 256MB --timeout 60s
```
Comment (hacktobeer) [inline deploy.sh:109]: nodejs8 will be decommissioned Feb 2021. Any specific reason we should use 8?

File: modules/turbinia/templates/scripts/install-turbinia-worker.sh.tpl [hunk 3]
```diff
@@ -36,22 +36,22 @@ done
```
Comment (hacktobeer) [inline modules/turbinia/templates/scripts/install-turbinia-worker.sh.tpl:36]: this file can be deleted, we use dockerized turbinia now.

**Review by hacktobeer — COMMENTED** [review]
Have a look at the comments and I'll run an e2e to do a final check after.

File: deploy.sh (hunk 1 above)
Comment (aarontp) [reply deploy.sh]: Yeah, this output was a little confusing, but it was meant to indicate that it wasn't set at run time.  I updated it to make it more accurate.

File: deploy.sh (hunk 2 above)
Comment (aarontp) [reply deploy.sh]: Good catch -- I think this came from me merging this from a PR I had started before we upgraded.  Updated back to 10.

File: modules/turbinia/templates/scripts/install-turbinia-worker.sh.tpl (hunk 3 above)
Comment (aarontp) [reply modules/turbinia/templates/scripts/install-turbinia-worker.sh.tpl]: Cool, yeah, I wasn't sure if we still needed this or not.  Deleted.

**Review by aarontp — COMMENTED** [review]
PTAL, thanks!

**Review by hacktobeer — APPROVED** [review]

**Comment by berggren** [comment]
@aarontp can you resolve the conflicts? Then feel free to merge this.

**Pull request merged**